imports/client/components/Puzzle.tsx

```diff
@@ -21,14 +21,109 @@ import PuzzleAnswer from './PuzzleAnswer';
 import PuzzleDeleteModal from './PuzzleDeleteModal';
 import PuzzleModalForm, { PuzzleModalFormSubmitPayload } from './PuzzleModalForm';
 import TagList from './TagList';
+import {
+  SolvedPuzzleBackgroundColor, UnsolvedPuzzleBackgroundColor,
+  ExpectsNoAnswersPuzzleBackgroundColor,
+} from './styling/constants';
+import { mediaBreakpointDown } from './styling/responsive';
+
+const PuzzleDiv = styled.div<{
+  expectsNoAnswers: boolean;
+  isUnsolved: boolean;
+  isSolved: boolean;
```

Comment:
These are mutually exclusive, right? Would we be happier if it was a single enum?

Reply:
Gave it another shot and I was able to extract this in a way that I liked well enough